firefox_ui_tests/remote/security/test_safe_browsing_warning_pages.py

```diff
@@ -0,0 +1,101 @@
+        # while between start up and adding the example urls to the db.
+        # hg.mozilla.org/mozilla-central/file/46aebcd9481e/browser/base/content/browser.js#l1194
+        time.sleep(3)
+
+    def test_warning_pages(self):
+        with self.marionette.using_context("content"):
+            for unsafe_page in self.urls:
+                # Load a test page and test the get me out button
+                self.marionette.navigate(unsafe_page)
+                self.check_get_me_out_of_here_button(unsafe_page)
+
+                # Load the test page again and test the report button
+                self.marionette.navigate(unsafe_page)
+                self.check_report_button(unsafe_page)
+
+                # Load the test page again and test the ignore warning button
+                self.marionette.navigate(unsafe_page)
+                self.check_ignore_warning_button(unsafe_page)
+
+    def check_get_me_out_of_here_button(self, unsafe_page):
+        button = self.marionette.find_element(By.ID, "getMeOutButton")
+
+        # Wait for the DOM to receive events
+        time.sleep(1)
+        button.click()
+
+        self.wait_for_condition(lambda mn: self.browser.default_homepage in mn.get_url())
+
+    def check_report_button(self, unsafe_page):
+        button = self.marionette.find_element(By.ID, "reportButton")
+
+        # Wait for the DOM to receive events
+        time.sleep(1)
+        button.click()
+
+        # Wait for page load so we can record and verify the url
+        self.wait_for_condition(lambda mn: mn.execute_script(
+            "return document.readyState == 'complete'"))
+        report_url = self.marionette.get_url()
+
+        with self.marionette.using_context('chrome'):
+            if 'its-a-trap' in unsafe_page:
+                url = self.marionette.execute_script("""
+                   Cu.import("resource://gre/modules/Services.jsm");
+                   return (Services.urlFormatter.formatURLPref("app.support.baseURL")
+                                                              + "phishing-malware");
+                   """)
+            else:
+                url = self.marionette.execute_script("""
+                   Cu.import("resource://gre/modules/Services.jsm");
+                   return (Services.urlFormatter.formatURLPref(
+                            "browser.safebrowsing.malware.reportURL") + arguments[0]);
+                   """, script_args=[unsafe_page])
+
+        # check that report_url matches the loaded url that we expect
+        self.assertEquals(report_url, self.browser.get_loaded_url(url))
+
+    def check_ignore_warning_button(self, unsafe_page):
+        button = self.marionette.find_element(By.ID, 'ignoreWarningButton')
+
+        # Wait for the DOM to receive events
+        time.sleep(1)
+        button.click()
+
+        self.wait_for_condition(expected.element_present(By.ID, 'main-feature'))
+        self.wait_for_condition(expected.element_stale(button))
```

Comment:
The ignore warning button doesn't seem to do much, at least, with current Nightly, apart from activating the safe browsing notification bar. Waiting for the "main-feature" element is the easy substitute for detecting the notification bar itself.